app/js/ZipManager.js

```diff
@@ -115,12 +119,32 @@ async function generateZip(projectId, zipfile) {
   zipfile.end()
 }
 
-async function exportProject(projectId) {
-  var zipfile = new yazl.ZipFile()
-  generateZip(projectId, zipfile) // generate zip file in background
-  return zipfile.outputStream
-}
+async function exportProject(projectId, res) {
+  const tmpdir = await fs.promises.mkdtemp(await fs.promises.realpath(os.tmpdir()) + path.sep);
```

Comment:
Not sure if this needs to be async, maybe it should be `const tmpdir = fs.mkdtempSync(fs.realpathSync(os.tmpdir()) + path.sep)`

Reply:
No, this is good. Better not to block the main thread while waiting for I/O

Reply:
The intention is that the `exportProject` promise gets rejected when the error is thrown, so it ends up being passed to `next` in the route handler.